### PR TITLE
[vcpkg] Add swig acquire

### PR DESCRIPTION
**Pull request by marekr**
SWIG is a interface compiler that allows binding C and C++ to scripting languages like Python, Ruby, Tcl and others. It's purely a compile-time tool to generate the bindings agaisnt your code.

I have a future port that will use this however, even end-user applications may find it handy to just vcpkg_find_package_acquire_swig for windows instead of needing a separate system install.

### CODE REVIEW

**Comment by Neumann-A** [comment]
also included in #11208

**Comment by Voskrese** [comment]
add SWIG_VERSION ?

```
  elseif(VAR MATCHES "SWIG")
    set(PROGNAME swig)
	set(SWIG_VERSION "4.0.2")
    set(PATHS "${DOWNLOADS}/tools/swig/swigwin-${SWIG_VERSION}")
    set(URL 
      "https://jaist.dl.sourceforge.net/project/swig/swigwin/swigwin-4.0.2/swigwin-${SWIG_VERSION}.zip"
      "https://sourceforge.net/projects/swig/files/swigwin/swigwin-4.0.2/swigwin-${SWIG_VERSION}.zip/download"
    )
    set(ARCHIVE "swigwin-${SWIG_VERSION}.zip")
    set(HASH b8f105f9b9db6acc1f6e3741990915b533cd1bc206eb9645fd6836457fd30789b7229d2e3219d8e35f2390605ade0fbca493ae162ec3b4bc4e428b57155db03d)
```

**Comment by marekr** [comment]
> add SWIG_VERSION ?
> 
> ```
>   elseif(VAR MATCHES "SWIG")
>     set(PROGNAME swig)
>     set(SWIG_VERSION "4.0.2")
>     set(PATHS "${DOWNLOADS}/tools/swig/swigwin-${SWIG_VERSION}")${SWIG_VERSION}
>     set(URL 
>       "https://jaist.dl.sourceforge.net/project/swig/swigwin/swigwin-${SWIG_VERSION}/swigwin-${SWIG_VERSION}.zip"
>       "https://sourceforge.net/projects/swig/files/swigwin/swigwin-${SWIG_VERSION}/swigwin-${SWIG_VERSION}.zip/download"
>     )
>     set(ARCHIVE "swigwin-${SWIG_VERSION}.zip")
>     set(HASH b8f105f9b9db6acc1f6e3741990915b533cd1bc206eb9645fd6836457fd30789b7229d2e3219d8e35f2390605ade0fbca493ae162ec3b4bc4e428b57155db03d)
> ```

Yea actually that'll be better

File: scripts/cmake/vcpkg_find_acquire_program.cmake [hunk 1]
```diff
@@ -345,6 +346,12 @@ function(vcpkg_find_acquire_program VAR)
     set(URL "https://github.com/aria2/aria2/releases/download/release-1.34.0/aria2-1.34.0-win-32bit-build1.zip")
     set(ARCHIVE "aria2-1.34.0-win-32bit-build1.zip")
     set(HASH 2a5480d503ac6e8203040c7e516a3395028520da05d0ebf3a2d56d5d24ba5d17630e8f318dd4e3cc2094cc4668b90108fb58e8b986b1ffebd429995058063c27)
+  elseif(VAR MATCHES "SWIG")
+    set(PROGNAME swig)
+    set(PATHS ${DOWNLOADS}/tools/swig/swigwin-4.0.2)
+    set(URL "https://sourceforge.net/projects/swig/files/swigwin/swigwin-4.0.2/swigwin-4.0.2.zip/download")
+    set(ARCHIVE "swigwin-4.0.2.zip")
+    set(HASH b8f105f9b9db6acc1f6e3741990915b533cd1bc206eb9645fd6836457fd30789b7229d2e3219d8e35f2390605ade0fbca493ae162ec3b4bc4e428b57155db03d)
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_find_acquire_program.cmake]:
```suggestion
elseif(VAR MATCHES "SWIG")
   set(VERSION 4.0.2)
   set(PROGNAME swig)
   if(CMAKE_HOST_WIN32)
       set(URL "https://sourceforge.net/projects/swig/files/swigwin/swigwin-${VERSION}/swigwin-${VERSION}.zip/download")
       set(ARCHIVE "swigwin-${VERSION}.zip")
       set(HASH b8f105f9b9db6acc1f6e3741990915b533cd1bc206eb9645fd6836457fd30789b7229d2e3219d8e35f2390605ade0fbca493ae162ec3b4bc4e428b57155db03d) 
       set(SUBDIR "swigwin-${VERSION}")
       set(PATHS "${DOWNLOADS}/tools/swig/${SUBDIR}/${SUBDIR}")
   else()
       #Not used
       set(_vfa_SUPPORTED TRUE)
       set(URL https://sourceforge.net/projects/swig/files/swig/swig-${VERSION}/swig-${VERSION}.tar.gz/download)
       set(ARCHIVE "swig-${VERSION}.tar.gz")
       set(HASH 05e7da70ce6d9a733b96c0bcfa3c1b82765bd859f48c74759bbf4bb1467acb1809caa310cba5e2b3280cd704fca249eaa0624821dffae1d2a75097c7f55d14ed) 
       set(SUBDIR "swig-${VERSION}")
       set(PATHS "${DOWNLOADS}/tools/swig/${SUBDIR}")
   endif()
   set(SOURCEFORGE_ARGS
       REPO swig
       FILENAME "${ARCHIVE}"
       SHA512 "${HASH}"
       NO_REMOVE_ONE_LEVEL
       WORKING_DIRECTORY "${DOWNLOADS}/tools/swig"
    )
```

**Comment by Voskrese** [comment]
@BillyONeal 


`Remove-Item : Cannot remove item D:\downloads\tools\msys2\msys64\usr\bin\msys-2.0.dll: Access to the path `

File: scripts/cmake/vcpkg_find_acquire_program.cmake (hunk 1 above)
Comment (marekr) [reply scripts/cmake/vcpkg_find_acquire_program.cmake]: What's up with the #not used path?

Comment (Neumann-A) [reply scripts/cmake/vcpkg_find_acquire_program.cmake]: I didn't test SWIG+ITK for the python bindings or it was using swig provided by the system somehow. Cannot remember anymore. Removed the comment

Comment (marekr) [reply scripts/cmake/vcpkg_find_acquire_program.cmake]: I can't apply your changes as its apparently out of date

File: scripts/cmake/vcpkg_find_acquire_program.cmake [hunk 2]
```diff
@@ -345,6 +346,13 @@ function(vcpkg_find_acquire_program VAR)
     set(URL "https://github.com/aria2/aria2/releases/download/release-1.34.0/aria2-1.34.0-win-32bit-build1.zip")
     set(ARCHIVE "aria2-1.34.0-win-32bit-build1.zip")
     set(HASH 2a5480d503ac6e8203040c7e516a3395028520da05d0ebf3a2d56d5d24ba5d17630e8f318dd4e3cc2094cc4668b90108fb58e8b986b1ffebd429995058063c27)
+  elseif(VAR MATCHES "SWIG")
+    set(PROGNAME swig)
+	set(SWIG_VERSION 4.0.2)
+    set(PATHS ${DOWNLOADS}/tools/swig/swigwin-${SWIG_VERSION})
+    set(URL "https://sourceforge.net/projects/swig/files/swigwin/swigwin-${SWIG_VERSION}/swigwin-${SWIG_VERSION}.zip/download")
+    set(ARCHIVE "swigwin-${SWIG_VERSION}.zip")
+    set(HASH b8f105f9b9db6acc1f6e3741990915b533cd1bc206eb9645fd6836457fd30789b7229d2e3219d8e35f2390605ade0fbca493ae162ec3b4bc4e428b57155db03d)
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_find_acquire_program.cmake:355]:
```suggestion
elseif(VAR MATCHES "SWIG")
   set(VERSION 4.0.2)
   set(PROGNAME swig)
   if(CMAKE_HOST_WIN32)
       set(URL "https://sourceforge.net/projects/swig/files/swigwin/swigwin-${VERSION}/swigwin-${VERSION}.zip/download")
       set(ARCHIVE "swigwin-${VERSION}.zip")
       set(HASH b8f105f9b9db6acc1f6e3741990915b533cd1bc206eb9645fd6836457fd30789b7229d2e3219d8e35f2390605ade0fbca493ae162ec3b4bc4e428b57155db03d) 
       set(SUBDIR "swigwin-${VERSION}")
       set(PATHS "${DOWNLOADS}/tools/swig/${SUBDIR}/${SUBDIR}")
   else()
       set(_vfa_SUPPORTED TRUE)
       set(URL https://sourceforge.net/projects/swig/files/swig/swig-${VERSION}/swig-${VERSION}.tar.gz/download)
       set(ARCHIVE "swig-${VERSION}.tar.gz")
       set(HASH 05e7da70ce6d9a733b96c0bcfa3c1b82765bd859f48c74759bbf4bb1467acb1809caa310cba5e2b3280cd704fca249eaa0624821dffae1d2a75097c7f55d14ed) 
       set(SUBDIR "swig-${VERSION}")
       set(PATHS "${DOWNLOADS}/tools/swig/${SUBDIR}")
   endif()
   set(SOURCEFORGE_ARGS
       REPO swig
       FILENAME "${ARCHIVE}"
       SHA512 "${HASH}"
       NO_REMOVE_ONE_LEVEL
       WORKING_DIRECTORY "${DOWNLOADS}/tools/swig"
    )
```

Comment (Neumann-A) [reply scripts/cmake/vcpkg_find_acquire_program.cmake:355]: hmm need to manually fix spaces if you apply the suggestion

**Comment by Voskrese** [comment]
@Neumann-A 

?

```

   if(CMAKE_HOST_WIN32)
...
       set(PATHS "${DOWNLOADS}/tools/swig/${SUBDIR}/${SUBDIR}")
...
```

`...\vcpkg\downloads\tools\swig\swigwin-4.0.2\swigwin-4.0.2\swig.exe`


enough of this, why so many extra nested folders?
` set(PATHS "${DOWNLOADS}/tools/swig/${SUBDIR}")`

`...\vcpkg\downloads\tools\swig\swigwin-4.0.2`

**Comment by Neumann-A** [comment]
Please fix it if you know how and where. It the strange zip extraction which creates the subfolders. Paths is just a hint where to find it.

**Comment by Voskrese** [comment]
```
   if(CMAKE_HOST_WIN32)
...
       set(PATHS "${DOWNLOADS}/tools/swig/${SUBDIR}/${SUBDIR}")
...
```

**Comment by Neumann-A** [comment]
Can be closed ITK PR was merged.

**Pull request closed without merging**